### PR TITLE
RFC: Support multiple JS.push events in LiveViewTest

### PR DESCRIPTION
**Pull request by rktjmp**
Currently LiveViewTest can't test views or components with something like:

```html
<button phx-click={JS.push("event_a") |> JS.push("event_b")}/>
```

This PR adds support for that by collecting all events and passing them through a reducer.

This PR retains the slightly unexpected current behaviour of merging `phx-value-x` attributes into `JS.push` data but only for the first `JS.push` event. It also preserves the current behaviour of using `phx-target` if present for `JS.push`, for all events.

### CODE REVIEW

File: lib/phoenix_live_view/test/client_proxy.ex [hunk 1]
```diff
@@ -313,58 +313,94 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
...
-              end
-
-            {view, targets, event, values, uploads}
+            # There may be multiple `push` events collected from JS commands,
+            # but we take care to only send the `value` and any `extra` values
+            # (collected from phx-value or form inputs) once with the first event.
+            #
+            # > I can't quite reckon when you might be able to send phx-value-x values
+            # > and uploads from a phx-click and also trigger JS events, but this
+            # > method retains any current behaviour, even if aberrant.
```
Comment (josevalim) [inline lib/phoenix_live_view/test/client_proxy.ex:322]: @rktjmp the upload comes an external parameter but I think it is reasonable to not allow it on JS.push. Can you please check the `phx-value` one? I am almost sure we don't include it in JS.push. If that's the case for both, I think we can move the values/upload handling to inside `maybe_js_events` (i.e. if it is a JS event, we don't attached it, otherwise we do).

**Comment by josevalim** [comment]
I dropped one small comment but it LGTM :+1: I was thinking about ways to remove the nested reduces but I couldn't come up with anything (for now).

File: lib/phoenix_live_view/test/client_proxy.ex (hunk 1 above)
Comment (rktjmp) [reply lib/phoenix_live_view/test/client_proxy.ex]: I went back to check and currently in 0.18.18, the additional values _are_ pulled in, eg:

```
    <div
      phx-click={
        JS.push("an-event")
        |> JS.push("an-event", value: %{"js" => true})
      }
      phx-value-key="something">
      button
    </div>
```

Gives

```
handle an-event: %{"key" => "something"}
handle an-event: %{"js" => true, "key" => "something"}
```

So I updated it to always send all values along and left the map(s) merging in the main block.

I left the uploads attached as I'm not sure if there are edge cases where it would be included.

**Comment by josevalim** [comment]
Beautiful. I will merge once ci passes.

**Comment by josevalim** [comment]
It is weird CI is not running. I will run tests locally then.

**Comment by rktjmp** [comment]
https://github.com/phoenixframework/phoenix_live_view/blob/b89c6086102a1808835354668f11fc0b639c4457/.github/workflows/ci.yml#L3-L7

`s/master/main`?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
